app/main.py

```diff
@@ -1,6 +1,15 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    a = []
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> any:
+        result = func(*args, **kwargs)
+        if result not in a:
+            a.append(result)
+            print("Calculating new result")
+        else:
+            print("Getting from cache")
+    return wrapper
```

Comment:
The `wrapper` function should return the `result` of the wrapped function `func`. Without returning the result, the decorated function will not produce any output, which is likely not the intended behavior.